packages/input/src/Input/Input.tsx

```diff
@@ -0,0 +1,157 @@
+import { Slot as RadixSlot } from "@radix-ui/react-slot";
+import { useComposedRefs } from "@telegraph/compose-refs";
+import clsx from "clsx";
+import React from "react";
+
+import { COLOR, SIZE } from "./Input.constants";
+
+type BaseRootProps = {
+  size?: keyof typeof SIZE.Container;
+  variant?: keyof typeof COLOR.Container.default;
+  error?: boolean;
+};
+
+type RootProps = BaseRootProps &
+  Omit<React.ComponentPropsWithoutRef<"input">, "size">;
+
+type RootRef = HTMLInputElement;
+
+type InternalProps = Omit<BaseRootProps, "error"> & {
+  state: "default" | "disabled" | "error";
+};
+
+type Required<T> = {
+  [P in keyof T]-?: T[P];
+};
+
+const InputContext = React.createContext<Required<InternalProps>>({
+  state: "default",
+  size: "2",
+  variant: "outline",
+});
+
+const Root = React.forwardRef<RootRef, RootProps>(
+  (
+    {
+      size = "2",
+      variant = "outline",
+      className,
+      disabled,
+      error,
+      children,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement>(null);
+    const composedRefs = useComposedRefs(forwardedRef, inputRef);
+
+    const state = disabled ? "disabled" : error ? "error" : "default";
+
+    return (
+      <InputContext.Provider value={{ size, variant, state }}>
+        <div
+          className={clsx(
+            "box-border flex items-center transition-all",
+            "border-2 border-solid text-gray-12 placeholder:text-gray-10",
+            COLOR.Container[state][variant],
+            SIZE.Container[size],
+          )}
+          // Focus the input when clicking on the container
+          onPointerDown={(event) => {
+            const target = event.target as HTMLElement;
+
+            // Make sure we're not clicking on an interactive element
+            if (target.closest("button, a")) {
+              event.preventDefault();
+              return;
+            }
+
+            const input = inputRef.current;
+            if (!input) return;
+
+            requestAnimationFrame(() => {
+              input.focus();
+            });
+          }}
```

Comment:
Does 2 things:
- Doesn't focus input when clicking on a button inside of the input
- Focuses input when clicking anywhere in the container, for example on the leading icon that isn't a button.